### DISCUSSION TITLE
Force motor to turn on when yaw command arrives.

### DISCUSSION THREAD

**Discussion by Marcox2111** [bitcraze/discussions#510] [Q&A] [answered]
Hi, I have a question. I'm trying to develop a different platform (in particular a blimp) using crazyflie firmware but there is a thing that I don't know how to manage.
When we give a thrust command to the crazyflie, some motors will turn on depending on the power_distribution function. But if I want that some motors start spinning by a positive yaw command it is impossible without some thrust. I don't know if it is clear but I'll make an example.

I want that two motors to start spinning when a thrust command is given but they should be off when I give a yaw command. 

The purpose is to maintain the height with helium and use thrust motor just to compensate a little bit, and the other two motor as a propeller used to go forward and steer. But I changed the power distribution in order to have just pitch and yaw on two motor and thrust on the other two have thrust.

Thanks in advance.

**Comment by hmllr** [comment, marked as answer]
In your controller you define a controlMode (as for example here: https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/controller_pid.c#L61). I think the cleanest solution would be to add your own mode in https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/interface/stabilizer_types.h#L172 and implement it in the power distribution code: https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/power_distribution_quadrotor.c#L119 
Hope this helps,
Hanna

**Reply by Marcox2111** [reply]
Really thanks, I will try and let you know

**Reply by Marcox2111** [reply]
Okay, thanks to your comment I found that the issue was this part of code: https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/controller_pid.c#L145 that stops every motor if the thrust is zero, now everything work as expected.
